pyro/contrib/funsor/infer/elbo.py

```diff
@@ -0,0 +1,52 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import weakref
+
+import pyro.ops.jit
+from pyro.infer import ELBO as OrigELBO
+from pyro.util import ignore_jit_warnings
+
+
+class ELBO(OrigELBO):
+
+    def _get_trace(self, *args, **kwargs):
+        raise ValueError("shouldn't be here!")
+
+    def differentiable_loss(self, model, guide, *args, **kwargs):
+        raise NotImplementedError("Must implement differentiable_loss")
+
+    def loss(self, model, guide, *args, **kwargs):
+        return self.differentiable_loss(model, guide, *args, **kwargs).detach().item()
+
+    def loss_and_grads(self, model, guide, *args, **kwargs):
+        loss = self.differentiable_loss(model, guide, *args, **kwargs)
+        loss.backward()
+        return loss.item()
+
+
+def make_jit_elbo(elbo_cls):
```

Comment:
Would it be more pythonic to implement this as a base class? I think the following might work:
```py
class Jit_ELBO(ELBO):
    def differentiable_loss(...):
        ...

class JitTrace_ELBO(Jit_ELBO, Trace_ELBO):
    pass
```